### PR TITLE
Implement retry logic for when events fail to send to Kinesis.

### PR DESCRIPTION
**Pull request by LATaylor-guardian**

### CODE REVIEW

File: app/com/gu/eventsforwarder/event/EventForwarder.scala [hunk 1]
```diff
@@ -18,27 +19,55 @@ object EventForwarder {
...
-        val request = new PutRecordsRequest().withStreamName(streamName).withRecords(recordBatch.asJava)
-        val putRecordsResult = client.putRecords(request)
+    } else if (putRecordsResult.getFailedRecordCount > 0) {
+      logger.warn(s"${putRecordsResult.getFailedRecordCount} failed to send. Retrying... attempt: $timesRetried")
 
-        if (putRecordsResult.getFailedRecordCount > 0)
-          logger.warn(s"${putRecordsResult.getFailedRecordCount} failed to send")
-        else
-          logger.info(s"Success. Sent ${putRecordsResult.getRecords.size} records.")
+      if (timesRetried == 0) Thread sleep 5000 else Thread sleep 5000 * (timesRetried + 1)
```
Comment (cb372) [inline app/com/gu/eventsforwarder/event/EventForwarder.scala]: This line can be simplified to just `Thread.sleep(5000 * (timesRetried + 1))`

File: app/com/gu/eventsforwarder/event/EventForwarder.scala [hunk 2]
```diff
@@ -18,27 +19,55 @@ object EventForwarder {
...
-      }
+    } else {
+      logger.info(s"Success. Sent ${putRecordsResult.getRecords.size} records.")
+    }
+  }
+
+  /* Sequence number does not exist in response on records that failed. Therefore, identify all the success requests and return those not successful from the request map */
+  private def identifyFailedRequests(putRecordsResult: PutRecordsResult, putRecordsRequestMap: Map[String, PutRecordsRequestEntry]): Map[String, PutRecordsRequestEntry] = {
+    val successfulRecordsBySequenceNumber = putRecordsResult.getRecords.asScala.toList
+      .filter(result => result.getSequenceNumber != null)
```
Comment (cb372) [inline app/com/gu/eventsforwarder/event/EventForwarder.scala]: Is `result.getSequenceNumber != null` a foolproof way to check that the record was successful?

Comment (LATaylor-guardian) [inline app/com/gu/eventsforwarder/event/EventForwarder.scala]: "Represents the result of an individual record from a PutRecords request. A record that is successfully added to a stream includes SequenceNumber and ShardId in the result. A record that fails to be added to the stream includes ErrorCode and ErrorMessage in the result."

**Comment by cb372** [comment]
👍

**Pull request merged**